src/Runner/PHPT/PhptTestCase.php

```diff
@@ -450,6 +461,48 @@ private function shouldTestBeSkipped(array $sections, array $settings): bool
         return false;
     }
 
+    /**
+     * @param array<non-empty-string, non-empty-string> $sections
+     */
+    private function shouldRunSkipIfInSubprocess(array $sections, string $skipIfCode): bool
+    {
+        if (isset($sections['INI'])) {
+            // to get per-test INI settings, we need a dedicated subprocess
+            return true;
+        }
+
+        $detector    = new SideEffectsDetector;
+        $sideEffects = $detector->getSideEffects($skipIfCode);
+
+        if ($sideEffects === []) {
+            return false; // no side-effects
+        }
+
+        foreach ($sideEffects as $sideEffect) {
+            if ($sideEffect === SideEffect::STANDARD_OUTPUT) {
```

Comment:
We could even report a SkipIf condition to be broken, when it does not contain a standard-out side-effect